common/test/acceptance/pages/studio/settings_advanced.py

```diff
@@ -126,3 +127,71 @@ def get_values(self, key_list):
             result_map[key] = val
 
         return result_map
+
+    def get_displayed_settings_names(self):
+        """
+        Returns all settings displayed on the advanced settings page/screen/modal/whatever
+        We call it 'name', but it's really whatever is embedded in the 'id' element for each field
+        """
+        query = self.q(css=SETTINGS_NAME_SELECTOR)
+        return [match.get_attribute('id') for match in query]
+
+    def get_expected_settings_names(self):
```

Comment:
It would be nicer if these were properties. E.g.
```
    @property
    def expected_settings_names(self):
    [...]
```

```
expected_fields = self.advanced_settings.expected_settings_names
```